brouter-core/src/main/java/btools/router/RoutingEngine.java

```diff
@@ -271,6 +275,109 @@ public void doRun( long maxRunningTime )
     }
   }
 
+  private void postElevationCheck(OsmTrack track) {
+	OsmPathElement lastPt = null;
+	OsmPathElement startPt = null;
+	short lastElev = Short.MIN_VALUE;
+	short startElev = Short.MIN_VALUE;
+	short endElev = Short.MIN_VALUE;
+	int startIdx = 0;
+	int endIdx = -1;
+	int dist = 0;
+    for ( int idx = 0; idx < track.nodes.size(); idx++ )
+    {
+      OsmPathElement n = track.nodes.get(idx);
+      if ( n.getSElev() == Short.MIN_VALUE && lastElev != Short.MIN_VALUE) {
+		// start one point before entry point to get better elevation results
+		if (idx > 1) 
+			startElev = track.nodes.get(idx-2).getSElev(); 
+		if (startElev == Short.MIN_VALUE) startElev = lastElev;
+		startIdx = idx;
+		startPt = lastPt;
+		dist = 0;
+		if (lastPt!= null) dist += n.calcDistance(lastPt);
+	  } else if (n.getSElev() != Short.MIN_VALUE && lastElev == Short.MIN_VALUE && startElev != Short.MIN_VALUE) {
+		// end one point behind exit point to get better elevation results
+		endElev = track.nodes.get(idx+1).getSElev(); //n.getSElev();

```

Comment:
This is the line that is triggering

`java.lang.IndexOutOfBoundsException: Index 18 out of bounds for length 18`

The relevant call stack:

```
        at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
        at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
        at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
        at java.base/java.util.Objects.checkIndex(Objects.java:372)
        at java.base/java.util.ArrayList.get(ArrayList.java:459)
        at btools.router.RoutingEngine.postElevationCheck(RoutingEngine.java:309)
        at btools.router.RoutingEngine.tryFindTrack(RoutingEngine.java:626)
        at btools.router.RoutingEngine.findTrack(RoutingEngine.java:453)
        at btools.router.RoutingEngine.doRun(RoutingEngine.java:177)
        at btools.server.RouteServer.run(RouteServer.java:274)
```

Looking at the code I see this can happen and I am surprised it was not trigger by other test cases.

Reply:
Great, thank you for that. 
I remember having a similar problem in the past. It was gone when I changed the rd5 tile. But I saved the wrong one and now reactivated it. But I couldn't find the corresponding ways for the error. And in this case it didn't bring an error too.